### DISCUSSION TITLE
Configuring vitest to use singleFork for some files, not others

### DISCUSSION THREAD

**Discussion by slhck** [vitest-dev/vitest#6438] [Q&A] [answered]
I have a suite of tests that come from a legacy Jest setup that used `--runInBand` for every file. The underlying reason is that in some tests, a binary is called which reserves a lockfile at a given location on-disk, and I cannot yet make it use a random/standalone lockfile for each test run. So I have to fall back to running some tests on a file-by-file basis.

```typescript
import { defineConfig } from 'vitest/config';

export default defineConfig({
  test: {
    pool: 'forks',
    poolOptions: {
      forks: {
        singleFork: true,
      }
    },
    sequence: {
      hooks: 'list',
    },
  },
});
```

This would be the default config, but for some known files I want to actually forego the `singleFork` option. How can I achieve that?

**Comment by slhck** [comment]
Answering myself to help others. Create a file `vitest.workspace.ts` and there, add:

```typescript
import { defineWorkspace } from 'vitest/config'

const PARALLEL_CAPABLE_TEST_FILES = [
  'test/something.unit.test.ts',
  'test/somethingElse.unit.test.ts',
];

export default defineWorkspace([
  {
    extends: './vitest.config.ts',
    test: {
      name: 'parallel',
      include: PARALLEL_CAPABLE_TEST_FILES,
    },
  },
  {
    extends: './vitest.config.ts',
    test: {
      name: 'sequential',
      include: ['test/**/*.test.ts'],
      exclude: PARALLEL_CAPABLE_TEST_FILES,
      poolOptions: {
        forks: {
          singleFork: true,
        }
      },
    },
  },
]);
```

You need to remove the `singleFork` option from the `vitest.config.ts` file, of course.

Then run the parallel tests only with (same for `sequential`):

```bash
npx vitest --project parallel
```

**Reply by AriPerkkio** [reply, marked as answer]
You can also define the `singleFork` in the workspace file. Then you can run all tests with a single `vitest` command.

```ts
import { defineWorkspace } from "vitest/config";

export default defineWorkspace([
  {
    test: {
      name: "parallel",
      include: ["**/first.test.ts", "**/second.test.ts"],
    },
  },
  {
    test: {
      name: "sequential",
      include: ["**/third.test.ts", "**/fourth.test.ts"],
      poolOptions: { forks: { singleFork: true } },
    },
  },
]);
```

Here I have 4 tests:

```ts
import { test } from "vitest";

test("testing", async () => {
  console.log(process.pid);
  await new Promise((r) => setTimeout(r, 1000));
});
```

https://github.com/user-attachments/assets/7c9ea87a-c590-481e-8c91-642c468b89cc

**Reply by slhck** [reply]
Thanks! You're right, I forgot to actually specify the config for those two types of tests.
